component/component.go

```diff
@@ -189,10 +189,3 @@ type CreateDefaultConfigFunc func() Config
 func (f CreateDefaultConfigFunc) CreateDefaultConfig() Config {
 	return f()
 }
-
-// InstanceID uniquely identifies a component instance
-type InstanceID struct {
```

Comment:
Can we add a 

```go
// Deprecated use [<ref to new type>]
type InstanceID = <ref to new type>
```

here?

Reply:
Can I?  That would make component dependent on pipelines... which is most definitely circular

Reply:
We can use an internal module and make this and the actual type aliases of a symbol in that internal module until we remove it from `component`:

```go
// package internal

type InstanceID struct { /* actual definition goes here *? } 
```

```go
// package component

// Deprecated: [v0.103.0] use pipelines.InstanceID
type InstanceID = internal.InstanceID
```

```go
// package pipelines

type InstanceID = internal.InstanceID
```

Reply:
I don't think this is going to work - 

Pipeline depends on internal, which depends on pipeline since InstanceID depends on pipeline.ID

If I move pipeline.ID into this internal package, we solve that problem but then the component.InstanceID shim doesn't work - that means component imports internal and then internal imports component (InstanceID contains component.ID and component.Kind).  

Open to any ideas to fixing this..

Side note - this internal package has to be collector/internal/<blah> since it needs to be accessible by collector/component and collector/pipeline.

Reply:
A solution here would be to also move `component.ID` and `component.Kind` to internal and make shims also for these two on `component`. It sounds like a pain, so let's first discuss the rest of the PR

Reply:
Right that would probably work... but I do question whether its worth it for stability concerns.  Moving component.ID to internal/ would make component dependent on the root collector module, thus adding that dependency to almost every other module.

I don't think the blast radius of this change is especially large for contrib and I think, in this case, it makes sense to keep this change as is and not try to provide backwards compatible import shims.